### ISSUE TITLE
mDNS discovery seems to fail for services that contain a digit in their name

### ISSUE THREAD

**Issue by robertklep** [athombv/apps-sdk-issues#284]
Android TV's provide a name service called `_androidtvremote2._tcp.local` which can be discovered using mDNS.

However, it seems like the digit in the name is preventing Homey from discovering the service:
```
{
  "type": "mdns-sd",
  "mdns-sd": {
    "name": "androidtvremote2",
    "protocol": "tcp"
  },
  "id": "{{txt.bt}}"
}
```
 Other services provided by the same TV (`_airplay`, `_hap`) are discovered just fine.

If I remove the 2 from the name, it seems like Homey (at some point) starts to actively query for the service:
```
192.168.23.8.5353 > 224.0.0.251.5353: [udp sum ok] 0 PTR (QM)? _androidtvremote._tcp.local. (45)
```
(192.168.23.8 being Homey)

It never seems to do this when the name contains a digit.

A developer [on the community forum](https://community.homey.app/t/how-to-perform-mdns-discovery/72784) seems to have had the same issue, but they worked around it by looking for a different service provided by the same device.

**Comment by FricoRico** [comment]
I upvoted because I experienced the issue myself.

Interestingly enough, given the following example also breaks the prior working `mdns-google-cast.json` discovery. I really have to remove `mdns-androidtvremote.json` in order for stuff to work, even if I have no references to the file. It seems once it's in the app manifest mdns-sd discovery breaks.

```
#mdns-androidtvremote.json

{
  "type": "mdns-sd",
  "mdns-sd": {
    "name": "androidtvremote2",
    "protocol": "tcp"
  },
  "id": "{{txt.bt}}"
}
```

```
#mdns-google-cast.json

{
  "type": "mdns-sd",
  "mdns-sd": {
    "name": "google-cast",
    "protocol": "tcp"
  },
  "id": "{{txt.id}}"
}
```

**Comment by lovethyresson** [comment]
Seeing the same thing but with hyphens.


```
{
  "type": "mdns-sd",
  "mdns-sd": {
    "name": "nibejpi-restapi",
    "protocol": "tcp"
  },
  "id": "{{txt.serialnumber}}"
}
``` 

and getting 
```
DiscoveryStrategy {
  _events: [Object: null prototype] {
    result: [ [Function (anonymous)], [Function (anonymous)] ]
  },
  _eventsCount: 1,
  _maxListeners: 0,
  type: 'mdns-sd',
  [Symbol(kCapture)]: false,
  [Symbol(result)]: {}
}
```

dns-sd output

```
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
14:24:11.732  Add        3  15 .                    _tcp.local.          _spotify-connect
14:24:11.732  Add        3  15 .                    _tcp.local.          _smb
14:24:11.732  Add        3  15 .                    _tcp.local.          _device-info
14:24:11.732  Add        3  15 .                    _udp.local.          _trel
14:24:11.732  Add        3  15 .                    _udp.local.          _meshcop
14:24:11.733  Add        3  15 .                    _tcp.local.          _srpl-tls
14:24:11.733  Add        3  15 .                    _tcp.local.          _mediaremotetv
14:24:11.733  Add        3  15 .                    _tcp.local.          _nibejpi-restapi
14:24:11.733  Add        3  15 .                    _tcp.local.          _ssh
14:24:11.733  Add        3  15 .                    _tcp.local.          _sonos

```

**Comment by robertklep** [comment]
> and getting

You're just logging the `DiscoveryStrategy` object. You can retrieve discovery results using [`getDiscoveryResults()`](https://apps-sdk-v3.developer.homey.app/DiscoveryStrategy.html#getDiscoveryResults), and since that list can initially be empty, you also need to subscribe to the [`result`](https://apps-sdk-v3.developer.homey.app/DiscoveryStrategy.html#event:result) event.

**Comment by RobinBol** [comment]
@robertklep which Homey can you reproduce this issue on?

My Homey Pro (Early 2023) has no problem with the discovery strategy as you mention in your first post:

```
2024-02-21T15:12:11.396Z [log] [App] Got result: DiscoveryResultMDNSSD {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  id: 'DC:E5:5B:51:1C:68',
  lastSeen: 2024-02-21T15:12:10.950Z,
  address: '192.168.0.43',
  host: 'Android-9.local',
  port: 6466,
  name: 'TV2',
  fullname: '',
  txt: { bt: 'DC:E5:5B:51:1C:68' },
  [Symbol(kCapture)]: false
}
2024-02-21T15:12:11.400Z [log] [App] Got result: DiscoveryResultMDNSSD {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  id: 'DC:E5:5B:91:C0:D1',
  lastSeen: 2024-02-21T15:12:10.952Z,
  address: '192.168.0.147',
  host: 'Android-4.local',
  port: 6466,
  name: 'TV1',
  fullname: '',
  txt: { bt: 'DC:E5:5B:91:C0:D1' },
  [Symbol(kCapture)]: false
}
```

It looks like I can reproduce on Homey Pro (2016—2019).

**Comment by RobinBol** [comment]
Apparently mDNS service names are not allowed to be longer than 15 characters: https://datatracker.ietf.org/doc/html/rfc6335#section-5.1. That is likely why it is not showing up on Homey Pro (2016—2019). WIll see if we can relax that limit a bit.

**Comment by robertklep** [comment]
@RobinBol it's been a while since I posted this issue, so I can't really remember on which model I tested on 😅

As for mDNS service name limits: various mDNS implementations allow for longer service names (usually only for the `.local` domain), perhaps that's why it works on HP2023 but doesn't on older models.

**Comment by RobinBol** [comment]
Thanks, yes it seems that avahi on Homey Pro (Early 2023) has no problem to discover these services. So I will relax these limits a bit to make the behavior of Homey Pro (2016—2019) similar.

**Comment by robertklep** [comment]
@RobinBol fwiw, I realised that the IP address shown in my initial issue report is from my old Homey.

**Comment by RobinBol** [comment]
@robertklep this fix is live in Homey Pro (2016—2019) v10.0.8
